File: SelectSort.go
package SortAlgorithmDemo

func selectsort(data []int) {
    for i := 0; i < len(data) - 1; i++ {
        var min int = i;
        var temp int = data[i];
        for j := i + 1; j < len(data); j++ {
            if (data[j] < temp){
                min = j;
                temp = data[j];
            }
        }
        if min != i {
            swap(data, min, i);
        }
    }
}

func swap(data []int, a int, b int){
    var temp int = data[a];
    data[a] = data[b];
    data[b] = temp;
}

File: lua/SelectSort.lua
selectsort = {}

function swap(data, a, b)
    local temp = data[a]
    data[a] = data[b]
    data[b] = temp
end

function selectsort.sort(data)
    for i=1, #(data) - 1 do
    for j=i+1, #(data) do
        if data[j] < data[i] then
            swap(data, i, j)
        end
    end
    end
    return data
end

return selectsort